File: leetcode-algorithms/007. Reverse Integer/solution.py
def reverse(x):
	"""
	:type x: int
	:rtype: int
	"""

	if x < 0:
		res = -1 * int(str(-x)[::-1])
	else:
		res = int(str(x)[::-1])
	if abs(res) > 0x7fffffff:
		return 0
	return res


if __name__ == '__main__':
	x = 1563847412
	print(reverse(x))

File: leetcode-algorithms/007. Reverse Integer/solution.cpp
#include <iostream>
using namespace std;


int reverse(int x) {
	long long res = 0;
	int flag = 0;
	if(x<0)
		flag = 1;
	x = abs(x);
	while(x)
	{
		res = res*10 + x % 10;
		x /= 10;
	}

	if(flag)
		res = res * -1;
    return (res<INT_MIN || res>INT_MAX) ? 0 : res;
	
    }

int main()
{
	int x = -123;
	cout<<reverse(x)<<endl;
	return 0;
}